src/NBitcoin/IBitcoinSerializable.cs

```diff
@@ -86,43 +85,62 @@ public static int GetSerializedSize(this IBitcoinSerializable serializable, Prot
+        public static void FromBytes(this IBitcoinSerializable serializable, byte[] bytes, ProtocolVersion version = ProtocolVersion.PROTOCOL_VERSION)
         {
-            network = network ?? Network.Main;
-
             var bms = new BitcoinStream(bytes)
             {
                 ProtocolVersion = version,
-                ConsensusFactory = network.Consensus.ConsensusFactory
+                ConsensusFactory = new DefaultConsensusFactory()
             };
+
             serializable.ReadWrite(bms);
         }
 
-        public static T Clone<T>(this T serializable, ProtocolVersion version = ProtocolVersion.PROTOCOL_VERSION, Network network = null) where T : IBitcoinSerializable, new()
+        public static void FromBytes(this IBitcoinSerializable serializable, byte[] bytes, ConsensusFactory consensusFactory, ProtocolVersion version = ProtocolVersion.PROTOCOL_VERSION)
         {
-            network = network ?? Network.Main;
+            var bms = new BitcoinStream(bytes)
+            {
+                ProtocolVersion = version,
+                ConsensusFactory = consensusFactory
+            };
+
+            serializable.ReadWrite(bms);
+        }
 
-            T instance = network.Consensus.ConsensusFactory.TryCreateNew<T>();
+        public static T Clone<T>(this T serializable, ProtocolVersion version = ProtocolVersion.PROTOCOL_VERSION) where T : IBitcoinSerializable, new()
+        {
+            T instance = new DefaultConsensusFactory().TryCreateNew<T>();
             if (instance == null)
                 instance = new T();
 
-            instance.FromBytes(serializable.ToBytes(version, network), version, network);
+            instance.FromBytes(serializable.ToBytes(version), version);
 
             return instance;
         }
 
-        public static byte[] ToBytes(this IBitcoinSerializable serializable, ProtocolVersion version = ProtocolVersion.PROTOCOL_VERSION, Network network = null)
+        public static T Clone<T>(this T serializable, ConsensusFactory consensusFactory, ProtocolVersion version = ProtocolVersion.PROTOCOL_VERSION) where T : IBitcoinSerializable, new()
         {
-            network = network ?? Network.Main;
+            T instance = consensusFactory.TryCreateNew<T>();
+            if (instance == null)
+                instance = new T();
+
+            instance.FromBytes(serializable.ToBytes(consensusFactory, version), consensusFactory, version);
+
+            return instance;
+        }
 
+        public static byte[] ToBytes(this IBitcoinSerializable serializable, ProtocolVersion version = ProtocolVersion.PROTOCOL_VERSION)
+        {
             using (var ms = new MemoryStream())
             {
                 var bms = new BitcoinStream(ms, true)
                 {
                     ProtocolVersion = version,
-                    ConsensusFactory = network.Consensus.ConsensusFactory
+                    ConsensusFactory = new DefaultConsensusFactory()
```

Comment:
I suggest doing this `ConsensusFactory = new DefaultConsensusFactory()` inside `BitcoinStream` constructor

Reply:
OK will do!